src/lib/api/core/request.ts

```diff
@@ -231,6 +229,23 @@ export const getRequestBody = (options: ApiRequestOptions): any => {
   return undefined;
 };
 
+const shouldRetry = (error: Error): boolean =>
+  !!(
+    axios.isAxiosError(error) &&
+    ((error.response?.status &&
+      [502, 503, 504].includes(error.response.status)) ||
+      (error.code &&
+        [
+          // Some of these codes aren't on `AxiosError` for some reason, see:
+          //   * https://github.com/axios/axios/issues/4894#issuecomment-2062064639
+          "ECONNREFUSED",
+          "ENOTFOUND",
+          "ENETUNREACH",
+          AxiosError.ECONNABORTED,
+          AxiosError.ERR_NETWORK,
+        ].includes(error.code)))
```

Comment:
Very thorough. This is making me do a double take on my own rust retry middleware.

By the way, my understanding of your implementation here is that retry logic is only in place for server side outages. If a client's network connection is unstable, we are not protecting against that. Is that correct?

Reply:
I think all, or at least most, of these `E...` errors could correspond to local network connection issues, so I was trying to protect against that.

Reply:
Oh! That's good to know.  I followed the github link to a discussion and saw this:

> for example, the error is ECONNREFUSED because the server isn't running

and assumed they were all very special types of server outages